### PR TITLE
fix: catching failed ready promise

### PR DESCRIPTION
**Pull request by odedhutzler**
### Description of the Changes

Added catch to a failed promise, and passed the error event

### CheckLists

- [x] changes have been done against master branch, and PR does not conflict
- [x] new unit / functional tests have been added (whenever applicable)
- [x] test are passing in local environment 
- [x] Travis tests are passing (or test results are not worse than on master branch :))
- [ ] Docs have been updated

### CODE REVIEW

File: src/player.js [hunk 1]
```diff
@@ -1129,6 +1129,8 @@ export default class Player extends FakeEventTarget {
     this._readyPromise = new Promise((resolve, reject) => {
       this._eventManager.listen(this, CustomEvents.TRACKS_CHANGED, resolve);
       this._eventManager.listen(this, Html5Events.ERROR, reject);
+    }).catch(e => {
+      this.dispatchEvent(e);
```
Comment (OrenMe) [inline src/player.js]: It looks like we are throwing the error event in multiple places, please verify that we only throw it once

File: src/player.js [hunk 2]
```diff
@@ -1129,7 +1129,7 @@ export default class Player extends FakeEventTarget {
     this._readyPromise = new Promise((resolve, reject) => {
       this._eventManager.listen(this, CustomEvents.TRACKS_CHANGED, resolve);
       this._eventManager.listen(this, Html5Events.ERROR, reject);
-    });
+    }).catch(() => {});
```
Comment (OrenMe) [inline src/player.js]: add a comment
```javascript
}).catch(() => {
//silence the promise rejection, error is handled by the error event
});
```

**Comment by dan-ziv** [comment]
@odedhutzler please fix tests

**Comment by odedhutzler** [comment]
@dan-ziv tests fixed.

File: test/src/player.spec.js [hunk 3]
```diff
@@ -278,20 +278,30 @@ describe("ready", function () {
         });
 
         it("should fail ready -> load", (done) => {
-          player.ready()
-            .catch((error) => {
-              error.type.should.be.equal('error');
+          player.addEventListener('error', (error) => {
```
Comment (OrenMe) [inline test/src/player.spec.js]: use ENUM

Comment (OrenMe) [reply test/src/player.spec.js]: in all tests

**Pull request merged**